src/main/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDao.java

```diff
@@ -63,10 +63,10 @@ public int insert(String userId, String service, CreateDraft newDraft) {
             keyHolder,
             new String[] {"id"}
         );
-        if (keyHolder.getKey() == null) {
-            throw new DataRetrievalFailureException("Unable to retrieve the generated key [null] or key not generated.");
+        if (keyHolder.getKey() != null) {
+            return keyHolder.getKey().intValue();
         }
```

Comment:
I think you might still get pinged maybe, perhaps something like?
```suggestion
		var key = keyHolder.getKey()
        if (key != null) {
            return key.intValue();
        }
```